parser/ParseBlockProperties.hpp

```diff
@@ -0,0 +1,218 @@
+   *          StorageBlockLayoutDescription message
+   * @param values list of parse strings representing the assignment of the property
+   **/
+  ParseBlockPropertyItem( int line_number, 
+                          int column_number,
+                          Property property,
+                          PtrList<ParseString>* values)
+                                  : ParseTreeNode(line_number, column_number),
+                                    property_(property),
+                                    values_(values) { }
+
+  std::string getName() const override {
+    return "BlockPropertyItem";
+  }
+
+  Property property() const {
+    return property_;
+  }
+
+  const ParseString* value() const {
+    return &*values_->begin();
+  }
+
+  const PtrList<ParseString>& values() const {
+    return *(values_.get());
+  }
+
+  /**
+   * @brief Gets a string version of the property which this pair describes
+   * 
+   * @return a string description of the property
+   */
+  std::string getPropertyString() const;
+
+  /**
+   * @brief Returns if the user specified to compress all attributes using a special keyword
+   * 
+   * @return true if this is a compression property and the value is all
+   */
+  bool compressAll() const;
+
+ protected:
+
+  void getFieldStringItems(
+      std::vector<std::string> *inline_field_names,
+      std::vector<std::string> *inline_field_values,
+      std::vector<std::string> *non_container_child_field_names,
+      std::vector<const ParseTreeNode*> *non_container_child_fields,
+      std::vector<std::string> *container_child_field_names,
+      std::vector<std::vector<const ParseTreeNode*>> *container_child_fields) const override {
+
+    inline_field_names->push_back("property");
+    inline_field_values->push_back(getPropertyString());
+
+    inline_field_names->push_back("value");
+
+    // pretty print multiple values as a list
+    if(values_->size() > 1) {
+      std::string str("(");
+      for(const ParseString& parseString : *values_) {
+        str += parseString.value() + ',';
+      }
+      str.erase(str.size() - 1, 1);
+      str += ")";
+      inline_field_values->push_back(str);
+    } else {
+      inline_field_values->push_back(values_->begin()->value());
+    }
+  }
+
+ private:
+
+  Property property_;
+  std::unique_ptr<PtrList<ParseString>> values_;
+
+  DISALLOW_COPY_AND_ASSIGN(ParseBlockPropertyItem);
+};
+
+/**
+ * @brief Contains user-specified physical properties of a table's blocks as BlockPropertyItem's
+ */
+class ParseBlockProperties : public ParseTreeNode {
+ public:
+
+  /**
+   * @brief Constructor.
+   *
+   * @param line_number Line number of the first token of this node in the SQL statement.
+   * @param column_number Column number of the first token of this node in the SQL statement.
+   **/
+  ParseBlockProperties( int line_number, 
+                        int column_number, 
+                        PtrList<ParseBlockPropertyItem>* properties)
```

Comment:
Missing doxygen for this parameter (should also make it clear that this object takes ownership of `properties`).